docs/v1/api.md

```diff
@@ -0,0 +1,230 @@
+# Service Catalog API
+
+This document proposes a Kubernetes-native service broker API. It is a formalization of [@krancour](http://github.com/krancour)'s [presentation](https://docs.google.com/presentation/d/1Fm3qG9zQ4R8ZbhtTObmQjO3UZYJ0ruYI4qr5w_gEKN8/edit?usp=sharing) that he gave at the sig-service-catalog meeting on 10/10/2016.
+
+# Motivations
+
+2 different demos were given at [10/3/2016’s sig-service-catalog meeting](https://www.youtube.com/watch?v=Kfar5Uw7CRg), both extremely promising. Each differed in quite a few ways, including how consumers used Kubernetes resources to communicate.
+
+This proposal aims to define a standard API, including a schema for Kubernetes-native resources, to:
+
+- Guide the continuing efforts shown in those demos
```

Comment:
I guess to me the larger goals are:

1.  Formalize an API around the service broker concept as a first-class, strongly typed, Kubernetes-like API in order to provide a uniform vocabulary to develop against
2.  Follow established patterns in the Kubernetes core API for globally-available assets and namespaced claims onto those assets
3.  Facilitate the development of Kubernetes-like controllers in support of the API

I try to avoid characterizing things as clunky in docs like this :)  Really, I have no problem at all with TPR, I just don't want to make the investment in making it better or live with its shortcomings for now and still have to make the investment later in a formal API and API server, when we could just make the investment now and get it out of the way while making things better for other groups that want to make first-class, robust APIs for Kube that live outside the core.